### PR TITLE
Improve build_opAssign documentation + minor refactoring

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: src/dmd/clone.d [hunk 1]
```diff
@@ -185,31 +185,70 @@ private bool needOpAssign(StructDeclaration sd)
...
 /******************************************
- * Build opAssign for struct.
- *      ref S opAssign(S s) { ... }
+ * Build opAssign for a `struct`.
  *
- * Note that `s` will be constructed onto the stack, and probably
- * copy-constructed in caller site.
+ * The generated `opAssign` function has the following signature:
+ *-----
+ *      ref S opAssign(S s)    // S is the name of the `struct`
```
Comment (jacob-carlborg) [inline src/dmd/clone.d]: 1. Please don't indent the code in the example
2. Only use three dashes for the code block

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/clone.d (hunk 1 above)
Comment (RazvanN7) [reply src/dmd/clone.d]: Done. Thanks!

**Review by jacob-carlborg — APPROVED** [review]

**Review by WalterBright — APPROVED** [review]

**Review by wilzbach — APPROVED** [review]

File: src/dmd/clone.d [hunk 2]
```diff
@@ -185,31 +185,70 @@ private bool needOpAssign(StructDeclaration sd)
...
+ *ref S opAssign(S s)    // S is the name of the `struct`
+ *---
  *
- * If S has copy copy construction and/or destructor,
- * the body will make bit-wise object swap:
- *          S __swap = this; // bit copy
- *          this = s;        // bit copy
- *          __swap.dtor();
- * Instead of running the destructor on s, run it on tmp instead.
+ * The opAssign function will be built for a struct `S` if the
```
Comment (andralex) [inline src/dmd/clone.d]: opAssign in backquotes

File: src/dmd/clone.d [hunk 3]
```diff
@@ -185,31 +185,70 @@ private bool needOpAssign(StructDeclaration sd)
...
+ * following constraints are met:
+ *
+ * 1. `S` does not have an identity `opAssign` defined.
```
Comment (andralex) [inline src/dmd/clone.d]: "identity" is defined without being used

**Pull request merged**